sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeComposeScreen.kt

```diff
@@ -1,29 +1,64 @@
 package co.nimblehq.sample.compose.ui.screens.home
 
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.fillMaxSize
+import android.widget.Toast
+import androidx.compose.foundation.layout.*
 import androidx.compose.material.CircularProgressIndicator
 import androidx.compose.material.Scaffold
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
 import co.nimblehq.sample.compose.R
+import co.nimblehq.sample.compose.lib.IsLoading
 import co.nimblehq.sample.compose.model.UiModel
+import co.nimblehq.sample.compose.ui.AppDestination
 import co.nimblehq.sample.compose.ui.screens.AppBar
-import co.nimblehq.sample.compose.ui.theme.ComposeSampleTheme
+import co.nimblehq.sample.compose.ui.theme.ComposeTheme
+import co.nimblehq.sample.compose.ui.userReadableMessage
 
 // TODO: Rename to 'HomeScreen'
 @Composable
```

Comment:
@luongvo I found some interesting pattern from Tivi project. They have another composable function to call the `private HomeScreen` for encapsulating the viewModel so when calling the `HomeScreen` from the `AppNavigation`, it will not see the `viewModel` property and cannot pass any value to it. What do you think?

Here it is what the function looks like:
```kotlin
@Composable
fun HomeScreen(
    navigator: (destination: AppDestination) -> Unit
) {
    HomeScreen(
        viewModel = hiltViewModel(),
        navigator = navigator
    )
}
```

And we can make the `fun HomeScreen` at line:23 to be private.

Reply:
@lydiasama As discussed [here](https://nimble-co.slack.com/archives/G41UCTGHL/p1671596317788699), maybe we can investigate it later in https://github.com/nimblehq/android-templates/issues/354 when actually working in testing 🤔